### ISSUE TITLE
First Order DataGenerator Prototyp erweitern

### ISSUE THREAD

**Issue by AshsGit11** [Cortys/SEP-Probly-WS2526#284]
Link zum Branch: https://github.com/HaoLMU/ProblyPros/tree/AprotypeFOGenerator
kann jetzt:
Kann Jetzt
-  p(Y|X=x) pro Sample aus Dataset/DataLoader rechnen
- Logits ->Softmax, konvertiert Logits zu Wks
- (Optional/Lösung für Unsicherheit) AutoModus erkennt bereits vorbereitete Wks und nutzt sie direkt
-  optionaler output_transform für Anpassungen
- device, CPU/GPU-Support; setzt das Modell in eval() (zum Sicherstellen das die Informationenne alle auf dem selben device sind)
-  einfache, abschaltbare Konsolen Progressanzeige (prints)
- Speichern: schreibt Verteilungen als JSON mit meta (z. B. model_name)
- (NOCH NICHT GETESTET) Laden: liest JSON und rekonstruiert Mapping index -> Wahrscheinlichkeitsvektor

**Comment by AshsGit11** [comment]
Commit that changed previous verison of simpler Protoype: https://github.com/HaoLMU/ProblyPros/commit/1fd0bc178ac6ba254cfc941ae9dc4b68072fa7d6
